package/kedro_viz/server.py

```diff
@@ -398,72 +425,108 @@ def format_pipeline_data(
+            if _is_dataset_param(dataset_full_name):
+                _add_parameter_data_to_node(dataset_full_name, task_id)
 
         for data_set in node.outputs:
-            namespace = data_set.split("@")[0]
-            namespace_to_layer[namespace] = dataset_to_layer.get(data_set)
-            namespace_id = _hash(namespace)
+            dataset_full_name = data_set.split("@")[0]
+            dataset_name_to_layer[dataset_full_name] = dataset_to_layer.get(data_set)
+            namespace_id = _hash(dataset_full_name)
             edge = {"source": task_id, "target": namespace_id}
             if edge not in edges_list:
                 edges_list.append(edge)
-            namespace_tags[namespace].update(node.tags)
+            dataset_name_tags[dataset_full_name].update(node.tags)
             node_dependencies[task_id].add(namespace_id)
+
     # Parameters and data
-    for namespace, tag_names in sorted(namespace_tags.items()):
-        is_param = _is_namespace_param(namespace)
-        node_id = _hash(namespace)
+    for dataset_full_name, tag_names in sorted(dataset_name_tags.items()):
+        dataset_modular_pipelines = []
+        if "." in dataset_full_name:
+            # The last part of the namespace is the actual name of the dataset
+            # e.g. in pipeline1.data_science.a, "pipeline1.data_science" indicates
+            # the modular pipelines and "a" the name of the dataset.
+            dataset_namespace = dataset_full_name.rsplit(".", 1)[0]
+            dataset_modular_pipelines = _expand_namespaces(dataset_namespace)
+        is_param = _is_dataset_param(dataset_full_name)
+        node_id = _hash(dataset_full_name)
 
         _JSON_NODES[node_id] = {
             "type": "parameters" if is_param else "data",
-            "obj": _get_dataset_data_params(namespace),
+            "obj": _get_dataset_data_params(dataset_full_name),
         }
-        if is_param and namespace != "parameters":
+        if is_param and dataset_full_name != "parameters":
             # Add "parameter_name" key only for "params:" prefix.
-            _JSON_NODES[node_id]["parameter_name"] = namespace.replace("params:", "")
+            _JSON_NODES[node_id]["parameter_name"] = dataset_full_name.replace(
+                "params:", ""
+            )
 
         if node_id not in nodes:
             nodes[node_id] = {
                 "type": "parameters" if is_param else "data",
                 "id": node_id,
-                "name": _pretty_name(namespace),
-                "full_name": namespace,
+                "name": _pretty_name(dataset_full_name),
+                "full_name": dataset_full_name,
                 "tags": sorted(tag_names),
-                "layer": namespace_to_layer[namespace],
+                "layer": dataset_name_to_layer[dataset_full_name],
                 "pipelines": [pipeline_key],
+                "modular_pipelines": dataset_modular_pipelines,
```

Comment:
I think we discussed that people can technically name their dataset `foo.bar` with `foo`  & `bar` not being modular pipeline, so we need to check whether they are in the valid modular pipelines list (which comes from the nodes)

Reply:
Ah okay, thanks for flagging that! I had that in mind for parameters, but will check for it for datasets as well 👍

Reply:
Done in https://github.com/quantumblacklabs/kedro-viz/pull/394/commits/50ca16c73d0e363ac69ae870e51368470fa915b1